pkg/app/two_pass_renderer.go

```diff
@@ -99,16 +103,22 @@ func (r *desiredStateLoader) twoPassRenderTemplateToYaml(inherited, overrode *en
 	if runtime.V1Mode {
 		var err error
 
-		finalEnv = initEnv
+		finalEnv, err = initEnv.Merge(overrode)
```

Comment:
Note: The end result of `finalEnv` is the same as before, right?

Reply:
not all. the end result of `finalEnv` is the result that `initEnv` and `overrode` merged. the values of `overrode` will override the values of `initEnv`.